python/algos/sequential/din/din_net.py

```diff
@@ -0,0 +1,156 @@
+#
+# Copyright 2022 DMetaSoul
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import torch 
+import metaspore as ms
+from ...layers import MLPLayer, DIEN_DIN_AttLayer, LRLayer
+
+
+
+class DIN(torch.nn.Module):
+    def __init__(self,
+                 use_wide=True,
+                 use_deep=True,
+                 din_embedding_dim=16,
+                 wide_embedding_dim=16,
+                 deep_embedding_dim=16,
+                 din_column_name_path=None,
+                 din_combine_schema_path=None,
+                 wide_column_name_path=None,
+                 wide_combine_schema_path=None,
+                 deep_column_name_path=None,
+                 deep_combine_schema_path=None,                 
+                 din_attention_hidden_layers=[16, 8],
+                 din_attention_hidden_activations='dice',
+                 din_attention_batch_norm=True,
+                 din_attention_dropout=0.1,
+                 din_hidden_layers=[32, 16],
+                 din_hidden_activations='dice',
+                 din_hidden_batch_norm=True,
+                 din_hidden_dropout=0.25,
+                 # din_seq_column_index_list[i]与din_target_column_index_list[i]做attention
+                 din_seq_column_index_list=[1, 3],
+                 din_target_column_index_list = [2, 4],
```

Comment:
Assert the length of din_seq_column_index_list and din_target_column_index_list must be same ?